Documentation/cmdref/cilium-agent.md

```diff
@@ -166,6 +166,7 @@ cilium-agent [flags]
       --mtu int                                       Overwrite auto-detected MTU of underlying network
       --nat46-range string                            IPv6 prefix to map IPv4 addresses to (default "0:0:0:0:0:FFFF::/96")
       --native-routing-cidr string                    Allows to explicitly specify the CIDR for native routing. This value corresponds to the configured cluster-cidr.
+      --neighbor-garbage-collect                      Garbage collects, on initialization, L2 entries created by a previous agent if neighbor discovery was enabled
```

Comment:
Do we really need this flag? Can we instead detect that previous instance of cilium-agent was running with neigh discovery on (e.g. by storing a bool and neigh discovery device in `/var/run/cilium`), and then if the neigh discovery is disabled remove permanent entries _of the neigh discovery device_?

Reply:
One other way to discovery and trigger this logic only _once_ would be when agent cleans up stale pinned maps, e.g. neigboring bpf map when the agent knows that its n.enableNeighDiscovery is false anyway. Then it could go into the neigh table and wipe all perm entries.

Reply:
Unfortunately we do as we can't guarantee that Cilium wasn't restarted twice and thus the map would have been deleted in the penultimate execution which would make it unavailable for us. Also I've marked this flag as disabled for now. I'm happy to discuss a better mechanism to always do it, regardless the value of the flag, by checking the map contents as a follow up.

Reply:
> Unfortunately we do as we can't guarantee that Cilium wasn't restarted twice

With the `/var/run/cilium` proposed approach you don't have this problem, as the state entry in fs would be removed only after you have removed all permanent entries.